naive/include/naive/ros_middleware.hpp

```diff
@@ -0,0 +1,56 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2022, PickNik, LLC.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+#pragma once
+
+#include <string>
+
+#include <rclcpp/rclcpp.hpp>
+#include <naive/incrementer.hpp>
+
+#include <std_msgs/msg/int64.hpp>
+
+class RosMiddleware : public Incrementer::MiddlewareHandle {
+public: 
+  RosMiddleware(const rclcpp::Node::SharedPtr node_handle, std::string in_topic,
+                const std::string& out_topic);
+  void registerCallback(Callback cb) override;
+  void publish(std_msgs::msg::Int64 msg) override;
+
+private:
+  static constexpr int QUEUE_SIZE = 10;
+  rclcpp::Node::SharedPtr node_handle_;
+  std::string in_topic_;
+  rclcpp::Subscription<std_msgs::msg::Int64>::SharedPtr subscriber_;
```

Comment:
I would prefer
```suggestion
  std::shared_ptr<rclcpp::Subscription<std_msgs::msg::Int64>> subscriber_;
```
but if you think it's more instructive to use the ros types, i'm ok with that.